### PR TITLE
Add water path diagnostics

### PR DESCRIPTION
**Pull request by AaronDonahue**
Introduce four new diagnostics to calculate water path.
    
    This commit establishes four diagnostics, each can be used to output
    the total water column path for the four water phases,
    1. vapor  - VapWaterPath
    2. liquid - LiqWaterPath
    3. ice    - IceWaterPath
    4. rain   - RainWaterPath

### CODE REVIEW

**Comment by AaronDonahue** [comment]
This PR is dependent on #1612 , so marking as WIP now, then once 1612 goes in we can switch this PR on.

**Comment by AaronDonahue** [comment]
Now that #1612 has been merged this should be nearly ready.  I'm keeping it at WIP so that I can add more testing.

File: components/scream/src/diagnostics/rain_water_path.hpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+#include "share/atm_process/atmosphere_diagnostic.hpp"
+#include "share/util/scream_common_physics_functions.hpp"
+#include "ekat/kokkos/ekat_subview_utils.hpp"
+
+namespace scream
+{
+
+/*
+ * This diagnostic will produce the potential temperature.
```
Comment (bartgol) [inline components/scream/src/diagnostics/rain_water_path.hpp:12]: Copy paste comment.

File: components/scream/src/diagnostics/ice_water_path.hpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+#include "share/atm_process/atmosphere_diagnostic.hpp"
+#include "share/util/scream_common_physics_functions.hpp"
+#include "ekat/kokkos/ekat_subview_utils.hpp"
+
+namespace scream
+{
+
+/*
+ * This diagnostic will produce the potential temperature.
```
Comment (bartgol) [inline components/scream/src/diagnostics/ice_water_path.hpp:12]: Copy paste comment.

**Review by bartgol — APPROVED** [review]

**Pull request merged**